### DISCUSSION TITLE
Does _app run on the Client or Server?

### DISCUSSION THREAD

**Discussion by SamRoehrich** [vercel/next.js#34305] [Help]
I ran into this behavior when trying to understand why SSR was not working in an application I work on. Some boilerplate code to get an idea of the issue.

`_app.js`

```
return (
    <AppWrapper>
          {children}
    </AppWrapper>
```

`AppWrapper.js`
```
const initialState = {loading: true, user: null}
const [state, setState] = useState(initialState)
useEffect(() => {
    const user = //some api call
    setState({ loading: false, user })
},[])
if (state.loading) return <Loading />
```

I realized that the if state.loading line was causing our pages using SSR to not work properly. The api calls would run on the server and get passed in as `pageProps` to the requested page, however if I viewed the source none of the page was getting built on the server. I am looking for some explanation on how this worked at all.

**Comment by GabenGar** [comment]
The entire page down to the file in the `pages` folder is built on the server. Hooks may vary, but the initial state in `useState()` is set on the server, but `useEffect()` only runs client-side.
So as far as the server concerned `state.loading` is always `true` during initial render.
